velox/exec/LocalPartition.cpp

```diff
@@ -377,28 +379,20 @@ void LocalPartition::addInput(RowVectorPtr input) {
       ContinueFuture future;
       auto reason = queues_[i]->enqueue(partitionData, &future);
       if (reason != BlockingReason::kNotBlocked) {
-        blockingReasons_[numBlockedPartitions_] = reason;
-        futures_[numBlockedPartitions_] = std::move(future);
-        ++numBlockedPartitions_;
+        blockingReasons_.push_back(reason);
+        futures_.push_back(std::move(future));
       }
     }
   }
 }
 
 BlockingReason LocalPartition::isBlocked(ContinueFuture* future) {
-  if (numBlockedPartitions_) {
-    --numBlockedPartitions_;
-    *future = std::move(futures_[numBlockedPartitions_]);
-    return blockingReasons_[numBlockedPartitions_];
-  }
-
-  if (noMoreInput_) {
-    for (const auto& queue : queues_) {
-      auto reason = queue->isFinished(future);
-      if (reason != BlockingReason::kNotBlocked) {
-        return reason;
-      }
-    }
+  if (!futures_.empty()) {
```

Comment:
So my understanding is to unblock an operator once any one of the partition queues becomes ready to consume or closed? Does have to be woken up one by one?

Reply:
This change allows the LocalPartition operator to pass all futures to the driver which will wake up LP operator only after all these future complete. Before this change we were passing one future at a time, which is inefficient.